Mock Helper incorrectly has assert for fixed or passed in strings

```
>       assert len(error_message) == len(self._defaults['error_message']['errorMessage'])
E       AssertionError: assert 256 == 35
E        +  where 256 = len(<nifake.session.c_char_Array_256 object at 0x04175CB0>)
E        +  and   35 = len('The answer to the ultimate question')
```

We should not be asserting the length of the string is equal to the length of the allocated size. We should be asserting the length of the allocated error_message is what we expect it to be which is 256 in this case.

--- a/build/templates/mock_helper.py.mako
+++ b/build/templates/mock_helper.py.mako
@@ -66,7 +66,6 @@ ivi_dance_size_param = helper.find_size_parameter(ivi_dance_param, params)
         if self._defaults['${func_name}']['${p['name']}'] is None:
             raise MockFunctionCallError("${c_function_prefix}${func_name}", param='${p['name']}')
 %       if p['is_buffer']:
-        assert len(${p['python_name']}) == len(self._defaults['${func_name}']['${p['name']}'])
         for i in range(len(${p['python_name']})):
             ${p['python_name']}[i] = self._defaults['${func_name}']['${p['name']}'][i]
 %       else:

--- a/generated/nidcpower/tests/mock_helper.py
+++ b/generated/nidcpower/tests/mock_helper.py
@@ -220,17 +220,14 @@ class SideEffectsHelper(object):
             return self._defaults['FetchMultiple']['return']
         if self._defaults['FetchMultiple']['voltageMeasurements'] is None:
             raise MockFunctionCallError("niDCPower_FetchMultiple", param='voltageMeasurements')
-        assert len(voltage_measurements) == len(self._defaults['FetchMultiple']['voltageMeasurements'])
         for i in range(len(voltage_measurements)):
             voltage_measurements[i] = self._defaults['FetchMultiple']['voltageMeasurements'][i]
         if self._defaults['FetchMultiple']['currentMeasurements'] is None:
             raise MockFunctionCallError("niDCPower_FetchMultiple", param='currentMeasurements')
-        assert len(current_measurements) == len(self._defaults['FetchMultiple']['currentMeasurements'])
         for i in range(len(current_measurements)):
             current_measurements[i] = self._defaults['FetchMultiple']['currentMeasurements'][i]
         if self._defaults['FetchMultiple']['inCompliance'] is None:
             raise MockFunctionCallError("niDCPower_FetchMultiple", param='inCompliance')
-        assert len(in_compliance) == len(self._defaults['FetchMultiple']['inCompliance'])
         for i in range(len(in_compliance)):
             in_compliance[i] = self._defaults['FetchMultiple']['inCompliance'][i]
         if self._defaults['FetchMultiple']['actualCount'] is None:
@@ -357,12 +354,10 @@ class SideEffectsHelper(object):
             return self._defaults['MeasureMultiple']['return']
         if self._defaults['MeasureMultiple']['voltageMeasurements'] is None:
             raise MockFunctionCallError("niDCPower_MeasureMultiple", param='voltageMeasurements')
-        assert len(voltage_measurements) == len(self._defaults['MeasureMultiple']['voltageMeasurements'])
         for i in range(len(voltage_measurements)):
             voltage_measurements[i] = self._defaults['MeasureMultiple']['voltageMeasurements'][i]
         if self._defaults['MeasureMultiple']['currentMeasurements'] is None:
             raise MockFunctionCallError("niDCPower_MeasureMultiple", param='currentMeasurements')
-        assert len(current_measurements) == len(self._defaults['MeasureMultiple']['currentMeasurements'])
         for i in range(len(current_measurements)):
             current_measurements[i] = self._defaults['MeasureMultiple']['currentMeasurements'][i]
         return self._defaults['MeasureMultiple']['return']
@@ -475,7 +470,6 @@ class SideEffectsHelper(object):
             return self._defaults['error_message']['return']
         if self._defaults['error_message']['errorMessage'] is None:
             raise MockFunctionCallError("niDCPower_error_message", param='errorMessage')
-        assert len(error_message) == len(self._defaults['error_message']['errorMessage'])
         for i in range(len(error_message)):
             error_message[i] = self._defaults['error_message']['errorMessage'][i]
         return self._defaults['error_message']['return']
@@ -490,12 +484,10 @@ class SideEffectsHelper(object):
             return self._defaults['revision_query']['return']
         if self._defaults['revision_query']['instrumentDriverRevision'] is None:
             raise MockFunctionCallError("niDCPower_revision_query", param='instrumentDriverRevision')
-        assert len(instrument_driver_revision) == len(self._defaults['revision_query']['instrumentDriverRevision'])
         for i in range(len(instrument_driver_revision)):
             instrument_driver_revision[i] = self._defaults['revision_query']['instrumentDriverRevision'][i]
         if self._defaults['revision_query']['firmwareRevision'] is None:
             raise MockFunctionCallError("niDCPower_revision_query", param='firmwareRevision')
-        assert len(firmware_revision) == len(self._defaults['revision_query']['firmwareRevision'])
         for i in range(len(firmware_revision)):
             firmware_revision[i] = self._defaults['revision_query']['firmwareRevision'][i]
         return self._defaults['revision_query']['return']
@@ -508,7 +500,6 @@ class SideEffectsHelper(object):
         self_test_result.contents.value = self._defaults['self_test']['selfTestResult']
         if self._defaults['self_test']['selfTestMessage'] is None:
             raise MockFunctionCallError("niDCPower_self_test", param='selfTestMessage')
-        assert len(self_test_message) == len(self._defaults['self_test']['selfTestMessage'])
         for i in range(len(self_test_message)):
             self_test_message[i] = self._defaults['self_test']['selfTestMessage'][i]
         return self._defaults['self_test']['return']

--- a/generated/nidmm/tests/mock_helper.py
+++ b/generated/nidmm/tests/mock_helper.py
@@ -249,7 +249,6 @@ class SideEffectsHelper(object):
             return self._defaults['FetchMultiPoint']['return']
         if self._defaults['FetchMultiPoint']['readingArray'] is None:
             raise MockFunctionCallError("niDMM_FetchMultiPoint", param='readingArray')
-        assert len(reading_array) == len(self._defaults['FetchMultiPoint']['readingArray'])
         for i in range(len(reading_array)):
             reading_array[i] = self._defaults['FetchMultiPoint']['readingArray'][i]
         if self._defaults['FetchMultiPoint']['actualNumberOfPoints'] is None:
@@ -262,7 +261,6 @@ class SideEffectsHelper(object):
             return self._defaults['FetchWaveform']['return']
         if self._defaults['FetchWaveform']['waveformArray'] is None:
             raise MockFunctionCallError("niDMM_FetchWaveform", param='waveformArray')
-        assert len(waveform_array) == len(self._defaults['FetchWaveform']['waveformArray'])
         for i in range(len(waveform_array)):
             waveform_array[i] = self._defaults['FetchWaveform']['waveformArray'][i]
         if self._defaults['FetchWaveform']['actualNumberOfPoints'] is None:
@@ -436,7 +434,6 @@ class SideEffectsHelper(object):
             return self._defaults['ReadMultiPoint']['return']
         if self._defaults['ReadMultiPoint']['readingArray'] is None:
             raise MockFunctionCallError("niDMM_ReadMultiPoint", param='readingArray')
-        assert len(reading_array) == len(self._defaults['ReadMultiPoint']['readingArray'])
         for i in range(len(reading_array)):
             reading_array[i] = self._defaults['ReadMultiPoint']['readingArray'][i]
         if self._defaults['ReadMultiPoint']['actualNumberOfPoints'] is None:
@@ -460,7 +457,6 @@ class SideEffectsHelper(object):
             return self._defaults['ReadWaveform']['return']
         if self._defaults['ReadWaveform']['waveformArray'] is None:
             raise MockFunctionCallError("niDMM_ReadWaveform", param='waveformArray')
-        assert len(waveform_array) == len(self._defaults['ReadWaveform']['waveformArray'])
         for i in range(len(waveform_array)):
             waveform_array[i] = self._defaults['ReadWaveform']['waveformArray'][i]
         if self._defaults['ReadWaveform']['actualNumberOfPoints'] is None:
@@ -513,7 +509,6 @@ class SideEffectsHelper(object):
             return self._defaults['error_message']['return']
         if self._defaults['error_message']['errorMessage'] is None:
             raise MockFunctionCallError("niDMM_error_message", param='errorMessage')
-        assert len(error_message) == len(self._defaults['error_message']['errorMessage'])
         for i in range(len(error_message)):
             error_message[i] = self._defaults['error_message']['errorMessage'][i]
         return self._defaults['error_message']['return']
@@ -528,12 +523,10 @@ class SideEffectsHelper(object):
             return self._defaults['revision_query']['return']
         if self._defaults['revision_query']['instrumentDriverRevision'] is None:
             raise MockFunctionCallError("niDMM_revision_query", param='instrumentDriverRevision')
-        assert len(instrument_driver_revision) == len(self._defaults['revision_query']['instrumentDriverRevision'])
         for i in range(len(instrument_driver_revision)):
             instrument_driver_revision[i] = self._defaults['revision_query']['instrumentDriverRevision'][i]
         if self._defaults['revision_query']['firmwareRevision'] is None:
             raise MockFunctionCallError("niDMM_revision_query", param='firmwareRevision')
-        assert len(firmware_revision) == len(self._defaults['revision_query']['firmwareRevision'])
         for i in range(len(firmware_revision)):
             firmware_revision[i] = self._defaults['revision_query']['firmwareRevision'][i]
         return self._defaults['revision_query']['return']
@@ -546,7 +539,6 @@ class SideEffectsHelper(object):
         self_test_result.contents.value = self._defaults['self_test']['selfTestResult']
         if self._defaults['self_test']['selfTestMessage'] is None:
             raise MockFunctionCallError("niDMM_self_test", param='selfTestMessage')
-        assert len(self_test_message) == len(self._defaults['self_test']['selfTestMessage'])
         for i in range(len(self_test_message)):
             self_test_message[i] = self._defaults['self_test']['selfTestMessage'][i]
         return self._defaults['self_test']['return']

--- a/generated/nifake/tests/mock_helper.py
+++ b/generated/nifake/tests/mock_helper.py
@@ -129,7 +129,6 @@ class SideEffectsHelper(object):
             return self._defaults['GetAStringOfFixedMaximumSize']['return']
         if self._defaults['GetAStringOfFixedMaximumSize']['aString'] is None:
             raise MockFunctionCallError("niFake_GetAStringOfFixedMaximumSize", param='aString')
-        assert len(a_string) == len(self._defaults['GetAStringOfFixedMaximumSize']['aString'])
         for i in range(len(a_string)):
             a_string[i] = self._defaults['GetAStringOfFixedMaximumSize']['aString'][i]
         return self._defaults['GetAStringOfFixedMaximumSize']['return']
@@ -139,7 +138,6 @@ class SideEffectsHelper(object):
             return self._defaults['GetAStringWithSpecifiedMaximumSize']['return']
         if self._defaults['GetAStringWithSpecifiedMaximumSize']['aString'] is None:
             raise MockFunctionCallError("niFake_GetAStringWithSpecifiedMaximumSize", param='aString')
-        assert len(a_string) == len(self._defaults['GetAStringWithSpecifiedMaximumSize']['aString'])
         for i in range(len(a_string)):
             a_string[i] = self._defaults['GetAStringWithSpecifiedMaximumSize']['aString'][i]
         return self._defaults['GetAStringWithSpecifiedMaximumSize']['return']
@@ -241,7 +239,6 @@ class SideEffectsHelper(object):
             return self._defaults['ReadMultiPoint']['return']
         if self._defaults['ReadMultiPoint']['readingArray'] is None:
             raise MockFunctionCallError("niFake_ReadMultiPoint", param='readingArray')
-        assert len(reading_array) == len(self._defaults['ReadMultiPoint']['readingArray'])
         for i in range(len(reading_array)):
             reading_array[i] = self._defaults['ReadMultiPoint']['readingArray'][i]
         if self._defaults['ReadMultiPoint']['actualNumberOfPoints'] is None:
@@ -257,7 +254,6 @@ class SideEffectsHelper(object):
         a_number.contents.value = self._defaults['ReturnANumberAndAString']['aNumber']
         if self._defaults['ReturnANumberAndAString']['aString'] is None:
             raise MockFunctionCallError("niFake_ReturnANumberAndAString", param='aString')
-        assert len(a_string) == len(self._defaults['ReturnANumberAndAString']['aString'])
         for i in range(len(a_string)):
             a_string[i] = self._defaults['ReturnANumberAndAString']['aString'][i]
         return self._defaults['ReturnANumberAndAString']['return']
@@ -310,7 +306,6 @@ class SideEffectsHelper(object):
             return self._defaults['error_message']['return']
         if self._defaults['error_message']['errorMessage'] is None:
             raise MockFunctionCallError("niFake_error_message", param='errorMessage')
-        assert len(error_message) == len(self._defaults['error_message']['errorMessage'])
         for i in range(len(error_message)):
             error_message[i] = self._defaults['error_message']['errorMessage'][i]
         return self._defaults['error_message']['return']

--- a/generated/niswitch/tests/mock_helper.py
+++ b/generated/niswitch/tests/mock_helper.py
@@ -382,7 +382,6 @@ class SideEffectsHelper(object):
             return self._defaults['error_message']['return']
         if self._defaults['error_message']['errorMessage'] is None:
             raise MockFunctionCallError("niSwitch_error_message", param='errorMessage')
-        assert len(error_message) == len(self._defaults['error_message']['errorMessage'])
         for i in range(len(error_message)):
             error_message[i] = self._defaults['error_message']['errorMessage'][i]
         return self._defaults['error_message']['return']
@@ -397,12 +396,10 @@ class SideEffectsHelper(object):
             return self._defaults['revision_query']['return']
         if self._defaults['revision_query']['instrumentDriverRevision'] is None:
             raise MockFunctionCallError("niSwitch_revision_query", param='instrumentDriverRevision')
-        assert len(instrument_driver_revision) == len(self._defaults['revision_query']['instrumentDriverRevision'])
         for i in range(len(instrument_driver_revision)):
             instrument_driver_revision[i] = self._defaults['revision_query']['instrumentDriverRevision'][i]
         if self._defaults['revision_query']['firmwareRevision'] is None:
             raise MockFunctionCallError("niSwitch_revision_query", param='firmwareRevision')
-        assert len(firmware_revision) == len(self._defaults['revision_query']['firmwareRevision'])
         for i in range(len(firmware_revision)):
             firmware_revision[i] = self._defaults['revision_query']['firmwareRevision'][i]
         return self._defaults['revision_query']['return']
@@ -415,7 +412,6 @@ class SideEffectsHelper(object):
         self_test_result.contents.value = self._defaults['self_test']['selfTestResult']
         if self._defaults['self_test']['selfTestMessage'] is None:
             raise MockFunctionCallError("niSwitch_self_test", param='selfTestMessage')
-        assert len(self_test_message) == len(self._defaults['self_test']['selfTestMessage'])
         for i in range(len(self_test_message)):
             self_test_message[i] = self._defaults['self_test']['selfTestMessage'][i]
         return self._defaults['self_test']['return']

